modules/seqexec/server/src/main/scala/seqexec/server/SeqTranslate.scala

```diff
@@ -323,29 +324,37 @@ object SeqTranslate {
     private def flatOrArcTcsSubsystems(inst: Instrument): NonEmptySet[TcsController.Subsystem] =
       NonEmptySet.of(AGUnit, (if (inst.hasOI) List(OIWFS) else List.empty): _*)
 
+    private def tryWavelength(inst: Instrument, config: CleanConfig): F[Option[Wavelength]] =
+      extractWavelength(config) match {
+        case Left(x) => Logger[F].error(s"Cannot decode the wavelength for ${inst.label}") *> MonadError[F, Throwable].raiseError(new SeqexecFailure.Execution(s"Cannot decode the wavelength from the sequence $x"))
+        case Right(w) => w.some.pure[F]
+      }
```

Comment:
As discussed we'll need to check whether `Wavelength` is present at all.  If that's not an error then we'd return `None` in that case.

Reply:
I added code to handle the case of no wavelength defined in the sequence. Local tests with this case went ok.